graphql_compiler/global_utils.py

```diff
@@ -29,3 +31,16 @@ def is_same_type(left, right):
         return is_same_type(left.of_type, right.of_type)
     else:
         return False
+
+
+def generate_new_name(base_name: str, taken_names: Sequence[str]) -> str:
+    """Return a name based on the provided string that is not already taken.
+
+    This method tries the following names:
+    {base_name}_0, then {base_name}_1, etc.
+    and returns the first one that's not in taken_names
+    """
+    index = 0
+    while "{}_{}".format(base_name, index) in taken_names:
+        index += 1
+    return "{}_{}".format(base_name, index)
```

Comment:
Deduplicate the `.format()` calls, to ensure that the returned value is guaranteed not to have been present in `taken_names`.